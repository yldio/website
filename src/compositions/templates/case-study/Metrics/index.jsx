import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { Copy, H2 } from 'components/Typography';
import PageSection from 'components/PageSection';
import MetricGrid from '../MetricGrid/index';

const Metrics = ({ sub, main, content }) => (
  <Fragment>
    <PageSection>
      <H2>Metrics</H2>
      <Copy large>{main}</Copy>
      <Copy newLines>{sub}</Copy>
    </PageSection>
    <PageSection>
      <MetricGrid content={content} />
    </PageSection>
  </Fragment>
);

Metrics.propTypes = {
  sub: PropTypes.string.isRequired,
  main: PropTypes.string.isRequired,
  ...MetricGrid.propTypes
};

export default Metrics;
