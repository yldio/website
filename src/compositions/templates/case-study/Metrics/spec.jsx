import React from 'react';
import { shallow } from 'enzyme';
import Metrics from './index';

test('components/Logo snapshot', () => {
  const metrics = {
    sub: 'some text',
    main: 'some text here as well',
    content: [{ parameter: 'test', value: '20.00', change: '20%' }]
  };
  const wrapper = shallow(<Metrics {...metrics} />);

  expect(wrapper).toMatchSnapshot();
});
