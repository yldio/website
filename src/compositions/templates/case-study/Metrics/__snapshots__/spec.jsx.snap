// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Logo snapshot 1`] = `
<React.Fragment>
  <PageSection>
    <Typography__H2>
      Metrics
    </Typography__H2>
    <Typography__base
      large={true}
    >
      some text here as well
    </Typography__base>
    <Typography__base
      newLines={true}
    >
      some text
    </Typography__base>
  </PageSection>
  <PageSection>
    <MetricGrid
      content={
        Array [
          Object {
            "change": "20%",
            "parameter": "test",
            "value": "20.00",
          },
        ]
      }
    />
  </PageSection>
</React.Fragment>
`;
